plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -627,10 +627,23 @@ public function maybe_update_nonce_user_logged_out( $uid, $action ) {
 	public function cleanup_sessions() {
 		global $wpdb;
 
-		$wpdb->query( $wpdb->prepare( 'DELETE FROM %i WHERE session_expiry < %d', $this->_table, time() ) );
-
-		if ( class_exists( 'WC_Cache_Helper' ) ) {
-			WC_Cache_Helper::invalidate_cache_group( WC_SESSION_CACHE_GROUP );
+		// The initial query can be slow for big session tables, but it doesn't make sense to introduce an index covering session_expiry due to high cardinality.
```

Comment:
Do we know how high this is?  Any testing done to confirm what this would look like?

Reply:
Replicating locally: can be interesting, but results will differ - it's not a clustered environment, and a single user is using it.
Still low-hanging fruit and a bit more data on the base system performance - I'll take an extra mile to try it out.